components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Request.java

```diff
@@ -109,13 +101,156 @@ public boolean isMultipart() {
 
   @Override
   public String toString() {
-    // Excluding some members which could lead to very verbose logging
     return getClass().getSimpleName() + "{" +
         "action='" + action + '\'' +
-        ", requestUrl='" + requestUrl + '\'' +
         ", path='" + path + '\'' +
         ", parameters=" + parameters +
         ", payload=" + payload +
+        ", multipart=" + multipart +
         '}';
   }
+
+  //
+  // Attributes
+  //
+
+  /**
+   * Custom attributes to configure backing and logging.
+   */
+  public static class Attributes
+      extends AttributesMap
+  {
+    public Attributes() {
+      super(Maps.<String, Object>newHashMap());
```

Comment:
Mostly just for logging so logs show org.sonatype.nexus.repository.view.Request.Attributes instead of org.sonatype.nexus.common.collect.AttributeMap.

It could however use the default ctor, think that came later than when this was originally written.

I made these public, now that we have a builder pattern, in case someone wanted to prime the attributes with an object, instead of using the default helpers.  Otherwise it could be private again.  /me shrugs.